### ISSUE TITLE
Draw Coordinate System button in toolbar

### ISSUE THREAD

**Issue by AJJLagerweij** [xournalpp/xournalpp#2682]
Hi, I'm requesting help here as I can't figure out the proper way to do this using the 'toolbar.ini' file. This is not a bug report neither a feature request but simply a help me.

I'm trying to make a new toolbar configuration for the floating toolbar, and I would like to have the 'Draw Coordinate System' as a button, but I cannot find the correct naming for it. I've tried `DRAW_COORDINATE`, `DRAW_COORDINATES` and `DRAW_COORDIYNATESYSTEM`, but none of these seem to work. What is the correct name for it. Of course, I can add the `DRAW` button to the floating toolbar, but that would require me to go into a sub menu each time which is exactly what I want to avoid.
Here below is my toolbar.ini file.
```ini
## Xournal++ Toolbar configuration
## Here you can customize the Toolbars
# Delete this file to generate a new config file with default values
#
# Available buttons:
# File: NEW,SAVE,OPEN
#
# Edit: CUT,COPY,PASTE,SEARCH,UNDO,REDO,INSERT_NEW_PAGE
#
# Navigation: GOTO_FIRST,GOTO_BACK,GOTO_NEXT,GOTO_LAST,GOTO_NEXT_ANNOTATED_PAGE
#
# Zoom: ZOOM_OUT,ZOOM_IN,ZOOM_FIT,ZOOM_100,FULLSCREEN,PAIRED_PAGES
#
# Color: COLOR(0xffffff),COLOR(0xffff00),COLOR(0xff8000),COLOR(0xff00ff),COLOR(0x00ff00),COLOR(0x00c0ff),COLOR(0x808080),COLOR(0x008000),COLOR(0xff0000),COLOR(0x3333cc),COLOR(0x000000),COLOR_SELECT
#  Notice: This are the default Xournal colors, each other color in HEX can also be used, eg COLOR(0x12ABCF);
#
# Tools: ERASER,PEN,HILIGHTER,IMAGE,TEXT,IMAGE,SELECT,SELECT_REGION,SELECT_RECTANGLE,VERTICAL_SPACE,HAND
#  Notice: ERASER also has a drop down menu to select the eraser type, SELECT are all selection tools, with drop down menu
#
# Tool settings: SHAPE_RECOGNIZER,RULER,FINE,MEDIUM,THICK,SELECT_FONT
#
# Components: PAGE_SPIN,ZOOM_SLIDER,LAYER
#  PAGE_SPIN: The page spiner, incl. current page label
#  ZOOM_SLIDER: The zoom slider
#  LAYER: The layer dropdown menu
#
#

[BRAM TABLET]
toolbarTop1=SAVE,NEW,OPEN,SEPARATOR,CUT,COPY,PASTE,,SEPARATOR,PAGE_SPIN,GOTO_FIRST,GOTO_NEXT_ANNOTATED_PAGE,GOTO_LAST,INSERT_NEW_PAGE,DELETE_CURRENT_PAGE,SEPARATOR,FULLSCREEN,PRESENTATION_MODE,ZOOM_100,ZOOM_FIT,ZOOM_IN,ZOOM_OUT,ZOOM_SLIDER
toolbarLeft1=DEFAULT_TOOL,ERASER,TOOL_FILL,TEXT,MATH_TEX,IMAGE,SEPARATOR,SELECT_RECTANGLE,SELECT_OBJECT,SELECT_REGION,VERTICAL_SPACE,HAND,SEPARATOR,ROTATION_SNAPPING,GRID_SNAPPING
toolbarFloat1=PEN,HILIGHTER,FINE,MEDIUM,THICK
toolbarFloat2=SHAPE_RECOGNIZER,RULER,DRAW_RECTANGLE,DRAW_CIRCLE,DRAW_ARROW,DRAW_COORDINATE,DRAW_COORDINATES,DRAW_COORDIYNATESYSTEM
toolbarFloat3=COLOR(0x000000),COLOR(0xffffff),COLOR(0xffff00),COLOR(0xff8000),COLOR(0xff00ff),COLOR(0x00ff00)
toolbarFloat4=COLOR(0x00c0ff),COLOR(0x808080),COLOR(0x008000),COLOR(0xff0000),COLOR(0x3333cc),COLOR_SELECT
name=Bram Tablet
```

**P.S.** Proposal for an enhancement (Should I make an extra issue for that?)
Similarly, but less importantly I have the issue that the `ERASER` and `PEN` always has a dropdown menu as well. It would be nice to be able to place a specific version in the floating bar, or to place the eraser and pen button in the floating toolbar which uses the 'current' type of the eraser and pen. But this would require an extra button for the eraser and pen, a version with dropdown, in another toolbar such that the type of eraser and pen can be set. This would be an enhancement which would allow us to use the floating toolbar as a quick toolbar where the more advanced options are in the toolbars at the side.

**Comment by personalizedrefrigerator** [comment]
A custom plugin _might_ address your second point. If it doesn't have support for adding new toolbar buttons, it should! [If you're interested, this file contains the plugin API.](https://github.com/xournalpp/xournalpp/blob/master/src/plugin/luapi_application.h).

Regarding your main issue, I would (personally) use the View > Toolbars > Customize menu action to create a new toolbar layout. To find the name of toolbar items, check [ToolMenuHandler.cpp](https://github.com/xournalpp/xournalpp/blob/1b7182891b9ccb268d214a9032f88a2fc6804b7d/src/gui/toolbarMenubar/ToolMenuHandler.cpp#L412).

I hope this is helpful!

**Comment by AJJLagerweij** [comment]
Yes that was it, `DRAW_COORDINATE_SYSTEM` did the trick. Anyway the [ToolMenuHandler.cpp](https://github.com/xournalpp/xournalpp/blob/1b7182891b9ccb268d214a9032f88a2fc6804b7d/src/gui/toolbarMenubar/ToolMenuHandler.cpp) is a nice place to look for when defining a custom toolbar layout.

I figured out what I did wrong when using the View > Toolbars > Customize menu. In Dutch this is called Weergave > Werkbalken > Aanpassen. However There are to menu items called Aanpassen. One brings me to the place where I can 'rename' the toolbar layout and one where I can edit the actual layout. I only tried the first one, but should have tried both.
![photo_2021-01-20_13-57-58](https://user-images.githubusercontent.com/31255888/105165534-84839480-5b27-11eb-88eb-0dba82e12f27.jpg) Where can I propose a translation.

I'll have a look at the plugins at some time as well. But it will take me a while to get to that.
